include/hip/hcc_detail/hip_runtime_api.h

```diff
@@ -1387,35 +1450,36 @@ hipError_t hipMemcpyDtoDAsync(hipDeviceptr_t dst, hipDeviceptr_t src, size_t siz
  * hipMemcpyFromArrayAsync, hipMemcpy2DFromArrayAsync, hipMemcpyToSymbolAsync,
  * hipMemcpyFromSymbolAsync
  */
-hipError_t hipMemcpyToSymbol(const void* symbolName, const void* src, size_t sizeBytes,
-                             size_t offset __dparm(0), hipMemcpyKind kind __dparm(hipMemcpyHostToDevice));
+inline
+__attribute__((visibility("hidden")))
+hipError_t hipMemcpyToSymbol(const void* symbolName, const void* src,
+                             size_t sizeBytes, size_t offset __dparm(0),
+                             hipMemcpyKind kind __dparm(hipMemcpyHostToDevice)) {
+    if (!symbolName) return hipErrorInvalidSymbol;
 
+    hipDeviceptr_t dst = NULL;
+    hipGetSymbolAddress(&dst, (const char*)symbolName);
```

Comment:
HIP API calling another HIP API isn't a good idea. This should be modified to use an ihipGetSymbolAddress.